projects/igniteui-angular/src/lib/grids/common/events.ts

```diff
@@ -76,22 +84,47 @@ export interface IGridEditDoneEventArgs extends IBaseEventArgs {
      * Indicates if the editing cosists of adding a new row
      */
     isAddRow?: boolean;
-    /** 
+    /**
      * Optional
-     * Indicates if the new value would be valid. 
-     * It can be set to return the result of the methods for validation of the grid 
+     * Indicates if the new value would be valid.
+     * It can be set to return the result of the methods for validation of the grid
      */
     valid?: boolean;
 }
 
-/** 
+
+/**
  * Represents event arguments related to grid editing.
  * The event is cancelable
  * It contains information about the row and the column, as well as the old and nwe value of the element/cell
  */
 export interface IGridEditEventArgs extends CancelableEventArgs, IGridEditDoneEventArgs {
 }
 
+export interface IRowDataCancelableEventArgs extends IRowDataEventArgs, IGridEditEventArgs {
```

Comment:
@damyanpetev  as suggested in our conversation  now the IRowDataCancelableEventArgs extends IRowDataEventArgs and IGridEditEventArgs, but in this way in this interface we now introduce a new deprecated property **data**  is this seems ok to you .. or maybe we can extend IRowDataEventArgs after we clean up  the now deprecated props?

Reply:
We had not decided deprecating cellId in `IgridEditEventArgs` at this moment